yang/iproute2-ip-route.yang

```diff
@@ -37,19 +37,50 @@ module iproute2-ip-route {
             ipr2cgen:cmd-delete "ip route del";
             ipr2cgen:cmd-start;
             ipr2cgen:replace-on-update;
+            ipr2cgen:oper-cmd "ip route show table all";
+            // TODO IPv6 support: oper needs to also apply "ip -6 route show"
             description
                 "ip route details";
             key "prefix table metric tos";
             // NODE_SPEC
+            leaf type{
+                type enumeration {
+                    enum "unicast";
+                    enum "local";
+                    enum "broadcast";
+                    enum "multicast";
+                    enum "unreachable";
+                    enum "prohibit";
+                    enum "blackhole";
+                    enum "nat";
+                }
+            }
+
             leaf prefix{
-                type inet:ipv4-prefix;
+                type union{
+                    type inet:ip-prefix;
+                    type inet:ip-address;
+                    type enumeration {
+                        enum "default";
+                    }
+                }
                 ipr2cgen:value-only;
+                ipr2cgen:oper-arg-name "dst";
                 description
-                    "ipv4 prifix";
+                    "ipv4 prefix";
             }
             leaf table {
-                onmcli:key-default-val "254";
-                type uint16;
+                onmcli:key-default-val "default";
+                ipr2cgen:oper-default-val "254";
+                type union {
+                    type enumeration {
```

Comment:
can we add value for the enum, 

```yang
enum "default" {  value 254; },

```

Reply:
for example if use added default, and then change it 254, this will not trigger a change on sysrepo.

Reply:
tested?
That means we can't use the enums with ints, either this or that

Reply:
done